### PR TITLE
[DHNE]

### PR DESCRIPTION
**Pull request by Vera-200**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [x] If the PR is for a new model/paper, I've updated the example index [here](../README.md).

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -3,6 +3,8 @@
 # @Email  : tyzhao@bupt.edu.cn
 
 import argparse
+import os.path
+
```
Comment (dddg617) [inline main.py:7]: Do we really need these lines?

File: main.py [hunk 2]
```diff
@@ -19,3 +21,4 @@
     experiment = Experiment(model=args.model, dataset=args.dataset, task=args.task, gpu=args.gpu,
                             use_best_config=args.use_best_config, load_from_pretrained=args.load_from_pretrained)
     experiment.run()
+
```
Comment (dddg617) [inline main.py:24]: Ditto.

File: openhgnn/dataset/HypergraphDataset.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        pos_data = [self.edges[idx]]
+        neg_data = []
+        n_neg = 0
+        while(n_neg < self.num_neg_samples):
+            # warning !!! we need deepcopy to copy list
+            index = copy.deepcopy(self.edges[idx])
+            mode = np.random.rand()
+            if mode < self.pair_radio:
+                type_ = np.random.randint(3)
+                # 随机选择一个node
```
Comment (dddg617) [inline openhgnn/dataset/HypergraphDataset.py:98]: Change Chinese to English.

File: openhgnn/dataset/HypergraphDataset.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                node = np.random.randint(self.nums_type[type_])
+                index[type_] = node
+            else:
+                # 随机选择两个类型
```
Comment (dddg617) [inline openhgnn/dataset/HypergraphDataset.py:102]: Ditto.

File: openhgnn/dataset/HypergraphDataset.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    return H
+
+
+def generate_embeddings(edge, nums_type):
+    r"""
+    Args:
+        edge (_type_): 边的数量
+        nums_type (_type_): 节点类型数量
+    Returns:
+        _type_: _description_
```
Comment (dddg617) [inline openhgnn/dataset/HypergraphDataset.py:146]: Ditto.

File: openhgnn/experiment.py [hunk 6]
```diff
@@ -78,7 +79,7 @@ def __init__(self, model, dataset, task,
         if use_best_config:
             self.config = set_best_config(self.config)
         self.set_params(**kwargs)
-        print(self)
```
Comment (dddg617) [inline openhgnn/experiment.py:81]: Do we need to delete this line? @Zhanghyi

File: openhgnn/models/DHNE.py [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+        Returns
+        -------
+        tensor
+            The logits after DHNE training.
+        """
+        encodeds = []
+        decodeds = []
+        for i in range(3):
+            encoded = torch.tanh(self.encodeds[i].to(self.device)(input_ids[i].to(self.device)))
```
Comment (dddg617) [inline openhgnn/models/DHNE.py:83]: I remember that this "to(self.device)" has been done in the "trainerflow", so do we really need parameter "device"?

File: openhgnn/models/DHNE.py [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+        device : int
+            the device DHNE working on
+        encodeds :
+            the input of autoencoder (the first layer of DHNE)
+        decodeds :
+            the output of autoencoder (the first layer of DHNE)
+        hidden_layer:
+            the second layer of DHNE
+        ouput_layer :
+            the third layer of DHNE
```
Comment (dddg617) [inline openhgnn/models/DHNE.py:37]: The Parameters listed here should correspond to the parameters in the function "__init__". The encodeds, decodeds, hidden_layer, and output_layer should be listed in the descriptions or deleted.

File: openhgnn/experiment.py (hunk 6 above)
Comment (Zhanghyi) [reply openhgnn/experiment.py:81]: Dont delete it. This line prints experiment information.

File: openhgnn/output/DHNE/DHNE.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+​	drug/GPS/MovieLens/wordnet
+
+### Dataset
+
+Supported dataset: drug, GPS, MovieLens, wordnet
+
+The details of datasets can be seen in the graph below:
+
+![image-20221013115422414](C:\Users\81480\AppData\Roaming\Typora\typora-user-images\image-20221013115422414.png)
```
Comment (dddg617) [inline openhgnn/output/DHNE/DHNE.md:28]: I think there will be an error in this line, as in "github", we do not have this path. You may change it to a table in Markdown like [this](https://github.com/BUPT-GAMMA/OpenHGNN/blob/main/README_EN.md#supported-models-with-specific-task).

**Review by Zhanghyi — APPROVED** [review]

**Pull request merged**